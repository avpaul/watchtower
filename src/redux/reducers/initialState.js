import { ALL } from '../constants/fellowFilters';

export const INDEX_PAGE_SIZE_DEFAULT = 20;
export const INDEX_PAGE_SIZE_OPTIONS = ['5', '10', '19', '30', '50', '100'];

const genericAPIStateArray = {
  loading: false,
  data: [],
  error: null
};

const genericAPIStateObject = {
  loading: false,
  data: {},
  error: null
};

export default {
  emsDashboard: {
    fellowsSummary: {
      loading: false,
      summary: [],
      error: ''
    }
  },
  fellows: {
    pagination: {
      perPage: 25,
      page: 1,
      pages: 1,
      firstPageURL: '',
      nextPageURL: '',
      prevPageURL: '',
      finalPageURL: ''
    },
    fellows: [],
    summary: { fellowsofftrack: 0, gteWk5OffTrack: 0, fellowsontrack: 0 },
    loading: false,
    error: ''
  },
  visibilityFilter: ALL,
  fellowCountHistory: {
    countSummary: {},
    loading: false
  },
  fellowsProgress: {
    loading: false,
    data: {
      D0A: [],
      D0B: [],
      D0: []
    },
    error: null
  },
  fellowDevPulse: {
    loading: false,
    ratings: [],
    averageRatings: {
      communication: 'N/A',
      integration: 'N/A',
      quantity: 'N/A',
      quality: 'N/A',
      professionalism: 'N/A',
      initiative: 'N/A'
    },
    error: null
  },
  fellowLmsSubmissions: {
    loading: false,
    lmsSubmissions: {},
    error: null
  },
  manager: {
    loading: false,
    data: {},
    error: null
  },
  opsSummary: {
    data: {
      managers: {
        ttls: [],
        lfs: [],
        averageFellowsPerTtl: 0,
        averageFellowsPerLf: 0
      },
      fellowsCount: {
        Total: 0
      },
      locations: []
    },
    loading: false
  },
  managerFellows: {
    loading: false,
    managerFellows: {},
    error: ''
  },
  fellow: {
    loading: false,
    fellow: {},
    error: ''
  },
  d1Fellow: {
    loading: false,
    fellow: {},
    error: ''
  },
  notification: {
    loading: false,
    notification: [],
    error: ''
  },
  unreadnotification: {
    loading: false,
    unreadnotification: {},
    error: ''
  },
  readnotification: {
    loading: false,
    readnotification: {},
    error: ''
  },
  fellowsSummary: {
    loading: false,
    fellowsSummaryToday: {},
    fellowsSummaryTrend: {},
    data: {},
    error: ''
  },
  engineeringManagerSimsLeads: {
    loading: false,
    data: {
      managers: {},
      locations: []
    },
    error: null
  },
  updateManagerNotification: {
    loading: false,
    managerNotification: [],
    error: null
  },
  managerFellowsSummary: {
    loading: false,
    data: [],
    error: null
  },
  feedback: {
    loading: false,
    data: [],
    error: null
  },
  fellowFeedback: {
    feedback: [],
    fellowFeedback: {}
  },
  pipData: {
    data: {},
    loading: false,
    error: null
  },
  performanceData: {
    loading: false,
    data: {
      today: {},
      trend: {}
    },
    error: ''
  },
  fellowPipFeedback: {
    loading: false,
    feedback: [],
    error: null
  },
  fellowPrePipFeedback: {
    loading: false,
    feedback: [],
    error: null
  },
  pipDeactivation: {
    success: null
  },
  createProject: genericAPIStateObject,
  createProjectVacancies: genericAPIStateObject,
  allProjects: genericAPIStateArray,
  fetchProjectManagers: genericAPIStateArray,
  fetchProjectTechnologies: genericAPIStateArray,
  editSingleProject: genericAPIStateObject,
  singleProject: genericAPIStateObject,
  createCertificactionVacancies: genericAPIStateObject,
  addProjectManager: {
    manager: {}
  },
  addProjectTechnology: {
    technology: ''
  },
  fetchSlackChannels: genericAPIStateArray,
  cadreEngineers: {
    loading: false,
    cadreEngineers: [],
    error: ''
  },
  cadreVacancies: genericAPIStateObject,
  reports: {
    loading: false,
    error: '',
    data: [],
    meta: {
      page: 1,
      perPage: INDEX_PAGE_SIZE_DEFAULT,
      perPageOptions: INDEX_PAGE_SIZE_OPTIONS,
      total: 0
    }
  },
  allRoles: genericAPIStateArray,
  fetchAllRoles: genericAPIStateArray,
  fetchActiveRole: genericAPIStateArray,
  createRole: genericAPIStateArray,
  roleSkills: genericAPIStateArray,
  getAllVacancies: genericAPIStateObject,
  allCetifications: genericAPIStateArray,
  editCertification: genericAPIStateObject,
  certifications: genericAPIStateArray,
  getCertification: genericAPIStateObject,
  editProjectVacancies: genericAPIStateObject,
  deleteProjectVacancies: genericAPIStateObject,
  projectVacanciesOnFocus: {},
  deleteCertification: genericAPIStateObject,
  certificationOnFocus: {},
  certificationApplicants: genericAPIStateArray,
  certificationApplication: genericAPIStateObject
};
