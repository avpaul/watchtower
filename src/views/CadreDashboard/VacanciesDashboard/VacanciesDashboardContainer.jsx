import { connect } from 'react-redux';
import VacancyDashboard from './VacanciesDashboard';
import {
  getAllVacancies as getAllVacanciesAction,
  setProjectVacanciesOnFocus
} from '../../../redux/actionCreators/projectVacancyActions';

export const mapStateToProps = ({ getAllVacancies }) => ({
  getAllVacancies
});

export default connect(
  mapStateToProps,
  {
    getAllVacanciesAction,
    setProjectVacanciesOnFocus
  }
)(VacancyDashboard);
