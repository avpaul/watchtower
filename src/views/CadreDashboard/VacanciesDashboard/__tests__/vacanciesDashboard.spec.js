import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Thunk from 'redux-thunk';

import configureStore from 'redux-mock-store';
import initialState from '../../../../redux/reducers/initialState';
import VacancyDashboard from '../VacanciesDashboard';
import VacanciesDashboardContainer from '../VacanciesDashboardContainer';

describe('Vacancy Dashboard', () => {
  const defaultProps = {
    getAllVacanciesAction: jest.fn(),
    getAllVacancies: {
      ...initialState.getAllVacancies,
      data: {
        projectVacancies: [
          {
            project: {
              id: 7,
              name: 'Corrupti non.'
            },
            role: {
              id: 1,
              name: 'Engineer'
            },
            vacancies: [
              {
                id: 99,
                project_id: 7,
                project_role_id: 1,
                fellow_id: null,
                is_active: false
              },

              {
                id: 105,
                project_id: 7,
                project_role_id: 1,
                fellow_id: null,
                is_active: false
              }
            ],
            available_slots: 2
          }
        ],
        certificationVacancies: [
          {
            "certification": {
              "id": 1,
              "name": "Wava Wehner MD",
              "description": "Harum sit reiciendis sint delectus aspernatur omnis.,",
              "exclusive": true,
              "duration": 20,
              "0": {
                "id": 2,
                "certification_id": 1,
                "fellow_id": null,
                "is_active": false,
                "created_at": "2019-07-10 07:58:34",
                "updated_at": "2019-07-10 07:58:34",
                "certification": {
                  "id": 1,
                  "name": "Wava Wehner MD",
                  "description": "Harum sit molestias.",
                  "exclusive": true,
                  "duration": 20
                }
              }
            },
            "available_slots": 2
          }
        ]
      }
    }
  };
  /**
   * Creates an enzyme instance to test the VacancyDashboard component.
   *
   * @param propOverrides Used to edit the props passed to the component when being mounted
   * @returns { wrapper, props }
   */
  const setup = (isMounted = false) => {
    const mockStore = configureStore([Thunk]);
    const store = mockStore(initialState);

    const wrapper = isMounted
      ? mount(
        <Provider store={store}>
          <VacanciesDashboardContainer />
        </Provider>
      )
      : shallow(<VacancyDashboard {...defaultProps} />);
    return { wrapper };
  };

  it('renders correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
