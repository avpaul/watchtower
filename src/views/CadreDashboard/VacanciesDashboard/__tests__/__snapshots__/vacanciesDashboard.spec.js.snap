// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vacancy Dashboard renders correctly 1`] = `
<div
  className="vacancies-dashboard cadre__page"
>
  <Connect(AddVacanciesModal) />
  <Connect(DeleteVacanciesModal) />
  <div
    className="row"
  >
    <div
      className="col-9"
    >
      <Title
        subTitle=""
        title="0 Vacancies (0 Projects, 0 Certifications)"
      />
    </div>
    <div
      className="col-3"
    >
      <CadreMainButton
        buttonProps={
          Object {
            "data-target": "#addProjectVacanciesModal",
            "data-toggle": "modal",
            "onClick": [Function],
          }
        }
        label="ADD VACANCIES"
      />
    </div>
  </div>
  <PaginatedVacanciesDashboard />
</div>
`;
